apis/v1alpha1/shared_types.go

```diff
@@ -62,13 +62,18 @@ type RouteGateways struct {
 	// * All: Gateways in any namespace can use this route.
 	// * FromList: Only Gateways specified in GatewayRefs may use this route.
 	// * SameNamespace: Only Gateways in the same namespace may use this route.
+	//
+	// +optional
 	// +kubebuilder:validation:Enum=All;FromList;SameNamespace
 	// +kubebuilder:default=SameNamespace
 	Allow GatewayAllowType `json:"allow,omitempty"`
+
 	// GatewayRefs must be specified when Allow is set to "FromList". In that
 	// case, only Gateways referenced in this list will be allowed to use this
 	// route. This field is ignored for other values of "Allow".
+	//
 	// +optional
+	// +kubebuilder:validation:MaxItems=16
```

Comment:
Is 16 an appropriate maximum number of gatewayRefs?

Reply:
Good catch, this does feel like an appropriate limit.

Reply:
@robscott @bowei @jpeach @hbagdi can I get feedback on ^?

Reply:
I think 16 is higher but I will not object against it strongly.
 
I can't think of cases where a single route is admitted to more than 16 Gateways.
Even if someone does find a use-case for this, between all kinds of selectors and lists we have in the API, the debugging experience, or even the general UX, of the API would be bad.

There could be some Service Mesh scenarios that would use this. My general preference is to start with lower limits.

Reply:
@hbagdi thanks for the feedback. I reduced the maximum to 8. @howardjohn PTAL and let us know if you have issue with this maximum.

Reply:
For a service mesh use case I would imagine its either a couple gateways at most, or 1 Gateway per service at which point this would be well beyond 16 and instead be 1000s which wouldn't work before or after this change. Probably the former would be a much more sensible way to model it.

So this seems reasonable

Reply:
I see `MaxItems=16` throughout the API's. I want to avoid having this PR cause a regression, so I'm going to remove this change.